lib/puppet/configurer.rb

```diff
@@ -246,9 +267,15 @@ def run_internal(options)
   private :run_internal
 
   def send_report(report)
+    reportSendTime = thinmark {
+      Puppet::Transaction::Report.indirection.save(report, nil, :environment => Puppet::Node::Environment.remote(@environment)) if Puppet[:report]
+    }
+    # Includes the indirection save time in the last_run_summary.yaml
+    # and in the summary
+    report.add_times(:reporting, reportSendTime)
+    report.finalize_report
     puts report.summary if Puppet[:summarize]
     save_last_run_summary(report)
-    Puppet::Transaction::Report.indirection.save(report, nil, :environment => Puppet::Node::Environment.remote(@environment)) if Puppet[:report]
```

Comment:
This means the report is sent before it's finalized, which I think will omit important information from what we send.